### DISCUSSION TITLE
Can't convert custom panel to Avalonia 11 due to removed IPanel interface

### DISCUSSION THREAD

**Discussion by BAndysc** [AvaloniaUI/Avalonia#14364] [Q&A]
Hello,

I have coded a custom 'foldable' panel in Avalonia 0.10, used for toolbars which was collapsing into a flyout for out-of-bounds items, as seen in the video below:

https://github.com/AvaloniaUI/Avalonia/assets/5689666/334e8287-5d6a-4f62-a29f-8ef803e62366

It worked fine, unfortunately I struggle to upgrade the code to Avalonia 11, because the panel depends on a custom IPanel implementation, but this interface was removed in Avalonia 11:

https://github.com/BAndysc/TestAvaloniaFoldablePanel/blob/master/TestAvaloniaFoldablePanel/ToolBarControl.axaml.cs#L38

Currently the panel consists of two parts:
 * [`ToolBarPanel`](https://github.com/BAndysc/TestAvaloniaFoldablePanel/blob/master/TestAvaloniaFoldablePanel/ToolBarPanel.cs) a `Panel` which is responsible for actual layouting and moving controls back and forth between the panel and the flyout
 * [`ToolBarControl`](https://github.com/BAndysc/TestAvaloniaFoldablePanel/blob/master/TestAvaloniaFoldablePanel/ToolBarControl.axaml.cs) a wrapper [Templated control](https://github.com/BAndysc/TestAvaloniaFoldablePanel/blob/master/TestAvaloniaFoldablePanel/ToolBarControl.axaml), which in its template contains `ToolBarPanel`, a `Flyout` and a button to show the flyout. The trick is that this control also implements `IPanel`, but places the children into the `ToolBarPanel`. This way this panel can be used i.e. in ItemsControl:
 
 ```
<ItemsControl DockPanel.Dock="Top">
        <ItemsControl.ItemsPanel>
                <ItemsPanelTemplate>
                        <testAvaloniaFoldablePanel:ToolBarControl />
                </ItemsPanelTemplate>
        </ItemsControl.ItemsPanel>
</ItemsControl>
```
Now, my problem is: I can't change `ToolBarControl` to `Panel`, because it needs to have flyout, button as its children. But at the same time I want to use it as `ItemsControl.ItemsPanel` which requires a `Panel`, not `IPanel` now, so this is a big problem.

Any advice how to solve this issue? Of course it is okay to change to approach if it is needed, I just need the ability to use the panel as ItemsPanel in the ItemsControl

[The full code is available here](https://github.com/BAndysc/TestAvaloniaFoldablePanel) with two configurations - for Avalonia 0.10 and Avalonia 11:

**Comment by timunie** [comment]
ItemsPanel != ControlTemplate. I believe your other properties and functions may want to be a part of the ControlTemplate instead.
